src/gei/Commands/MigrateArchiveRepoCommand.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class MigrateArchiveRepoCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ISourceGithubApiFactory _sourceGithubApiFactory;
+        private readonly ITargetGithubApiFactory _targetGithubApiFactory;
+        private readonly IAzureApiFactory _azureApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+        private const int ARCHIVE_GENERATION_TIMEOUT_IN_HOURS = 10;
+        private const int CHECK_STATUS_DELAY_IN_MILLISECONDS = 10000; // 10 seconds
+        private const string GIT_ARCHIVE_FILE_NAME = "gitArchive.tar.gz";
+        private const string METADATA_ARCHIVE_FILE_NAME = "metadataArchive.tar.gz";
+
+        public MigrateArchiveRepoCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubApiFactory, ITargetGithubApiFactory targetGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider, IAzureApiFactory azureApiFactory) : base("migrate-archive-repo")
+        {
+            _log = log;
+            _sourceGithubApiFactory = sourceGithubApiFactory;
+            _azureApiFactory = azureApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
+            _targetGithubApiFactory = targetGithubApiFactory;
+
+            Description = "Generates migration archives, uploads them to Azure Blob Storage, then invokes the GitHub Migration APIs to migrate the repo and all repo data using those uploaded archives.";
+            Description += Environment.NewLine;
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT and GH_SOURCE_PAT env variables to be set. GH_SOURCE_PAT is optional, if not set GH_PAT will be used instead. This authenticates to the source GHES API.";
```

Comment:
```suggestion
            Description += "Note: Expects GH_PAT and GH_SOURCE_PAT env variables to be set. GH_SOURCE_PAT  authenticates to the source GHES API. GH_PAT authenticates with the target api";
```

Reply:
Updated in https://github.com/github/gh-gei/pull/221/commits/6930b2dbbd56bb7dfc4f94c428757c503c443cfb